Update experiment tracking docs page for Kedro-Viz version and improved readability

stichbury opened the pull request:
## Description
I changed mentions of versions of Kedro-Viz to correspond to the version in which the latest experiment tracking is introduced and working. There were mentions of 4.0 and also 6.0 but neither seemed ideal since the latest is 5.2.1, so that's what I changed them all to.

## Development notes
I also modified the introduction to read a bit better, and the wording of some of the page subsections.

I do still have a bit of a problem with this page though and am hoping that @tynandebold and @rashidakanchwala can help me work out what to do. This is it:

So there are a number of GIFs which correspond to screenshots of the demo, which is awesome. There's also some sample code to show you how to enable experiment tracking on spaceflights.

But sadly, the code doesn't correspond to the demo output, so if you follow it, you don't get all the nice plotly plots that you do if you look at the demo (or look at the GIFs on the page). This is confusing for a reader (well, for me at least, and I don't think I'm alone).

What can we do?

1. Highest effort: We could fix the code snippets to add all the necessary code so that someone could take the spaceflights starter, add in the code on this page, and then run it and see _EXACTLY_ what is in the GIFs. Spoiler: I tried this and I couldn't get it to look exactly the same (where is the code for demo kept so I can use it?)
2. We could fix the GIFs so that what the user sees on the docs page is exactly what they'll see if they copy the code as it is given
3. Lowest effort: We could remove the GIFs and just point people at the demo at the start of the documentation

There may be additional options here but these are the three that occur to me. I would prefer to do 1. since then the reader gets exactly what they see in the demo, but for that, I'll need some assistance to build a Kedro project that does actually do what the demo does (as my example doesn't).


## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes


<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2292"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

rashidakanchwala reviewed (commented):
The reason for the demo and spaceflights being different is that demo project is not based on the spaceflights project. I think it's just fake data but related actually to the OptimusAI ShuttleFactory project. 

For testing purposes, we have always used demo-project. Also for every feature, we add some sort of fake data to the demo project so we can test and show it as an example to the users. 

I think for the tutorial, we will need to create meaningful fake data for spaceflights ( the meaningful part is something I too am not good at as I am not exactly a data practitioner) 

but overall if we don't want to do work twice  in the future - that is creating examples for latest features of demo project and creating examples on spaceflights tutorial maybe we nee a spaceflights demo project.

stichbury commented:
Aha, thanks @rashidakanchwala ! That helps a lot!!

I think we should revise the docs to make that clear somehow. I'm still not sure which option to take though, but it needs to be clear that what you see is not what you get.

NeroOkwa reviewed (approved):
Looks great, nice work!

rashidakanchwala commented:
> Aha, thanks @rashidakanchwala ! That helps a lot!!
> 
> I think we should revise the docs to make that clear somehow. I'm still not sure which option to take though, but it needs to be clear that what you see is not what you get.

Can we have a note below the images that this 'This Image is for illustration purpose and is not related the spaceflight tutorial'

stichbury commented:
Hi @rashidakanchwala -- yes, that's pretty much what I've done in the last couple of graphics, thanks!

@tynandebold and @rashidakanchwala it's now ready for review. I've made a few changes:

* Update to some of the preliminary text to make it a bit less complex
* Updated gif files to replace them with screen grabs from the actual code of spaceflights + matplotlib pipeline (which I've built and put on my own repo) https://github.com/stichbury/kedro-projects/tree/main/spaceflights_with_experiment_tracking/kedro-experiment-tracking-tutorial
* Added clarification when the demo gif is used to make it clear it's not a reflection of the code.

You can see the built version of the docs when they've propagated over to stichbury.github.io (https://stichbury.github.io/visualisation/experiment_tracking.html) but they've not updated yet.

rashidakanchwala reviewed (approved):
thanks @stichbury -- it's great!! :)

tynandebold commented on docs/source/visualisation/experiment_tracking.md:
```diff
@@ -1,49 +1,66 @@
 
-This page describes the steps necessary to set up experiment tracking and access logged metrics, using the [spaceflights tutorial](../tutorial/spaceflights_tutorial.md).
+### Prerequisites
+We assume that you have already [installed Kedro](../get_started/install.md) and [Kedro-Viz](../visualisation/kedro-viz_visualisation.md). Before proceeding, ensure you're using Kedro-Viz `>=5.2.1` (you can confirm your Kedro-Viz version by running `kedro info`).
```
This is tricky. Experiment tracking worked back to version `4.1.1`, as you removed from somewhere else on the page.

If we want to state when plots came to the product, that was with the release of `5.0.0`.

I'm not sure what's correct to put here, though `5.2.1` doesn't feel right.

tynandebold replied:
This is in the `Prerequisites` section though, so isn't this is the highest-level requirement for getting experiment tracking in Viz, not just the most up to date?

stichbury replied:
It's the pre-requisites for the code tutorial, which is definitely 5.2.1 but I can make that clearer. And I can add a "what is in what version?" section to clarify.

stichbury replied:
@tynandebold Please check this latest commit https://github.com/kedro-org/kedro/pull/2292/commits/decfc71cd6dfd3fce79aa41bc6d1400e1d1f05a4 for the revisions I've put in to cover the versions and what are supported. Please can you confirm they are correct or make the necessary changes?

stichbury replied:
You can see the new copy here https://stichbury.github.io/visualisation/experiment_tracking.html

tynandebold commented on docs/source/visualisation/experiment_tracking.md:
```diff
@@ -251,22 +276,31 @@ confusion_matrix:
+After running the pipeline with `kedro run`, the plot is saved and you can see it in the experiment tracking panel when you execute `kedro viz`. Clicking on a plot expands it. When in comparison view, expanding a plot shows all the plots in that view for side-by-side comparison.
 
-![](../meta/images/expand-plot-comparison-view.gif)
+![](../meta/images/experiment-tracking-plots-comparison.png)
+![](../meta/images/experiment-tracking-plots-comparison-expanded.png)
```
Can we please add a space in between these images? To the eye they can be mistaken for one single image.

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/16869061/217313982-8912c125-6e89-4f40-91f7-e46a12b7e886.png">

```suggestion
![](../meta/images/experiment-tracking-plots-comparison.png)

![](../meta/images/experiment-tracking-plots-comparison-expanded.png)
```

tynandebold reviewed (changes requested):
This is great and is another massive improvement! Thank you.

I've left some comments and will request a couple of changes, the biggest one being the first image on the page needs to be updated, as the UI shown doesn't reflect the current state of Viz.

Lastly, I notice that there aren't any GIFs/images of the new metrics plots we released recently. Is that by design? I don't think it's a necessity they're in there, though it's a neat thing and our newest feature.

stichbury commented on docs/source/visualisation/experiment_tracking.md:
```diff
@@ -1,49 +1,66 @@
+
+Take a look at the [Kedro-Viz experiment tracking demo](https://demo.kedro.org/experiment-tracking) for further information.
+
+![](../meta/images/experiment-tracking_demo.gif)
```
Thanks @tynandebold  -- I'll need someone to help me with the GIF creation as it's not something I can do smoothly (I have made those new images). 

Shall we make a separate ticket to update the images to GIFs in the near future rather than hold back on the updates to make them in this PR? 

Or go for it and get the new graphics in before we make the release? I can share my code (it's on my repo https://github.com/stichbury/kedro-projects/tree/main/spaceflights_with_experiment_tracking/kedro-experiment-tracking-tutorial) if you can help me make the GIFs.

tynandebold replied:
I think it's fine to use the demo project for making the GIF for that first image.

Why can't you make the GIFs smoothly? Is it because you don't have metrics plot setup in your local project? Or is it a tech issue?

stichbury replied:
It's user error. I assume I need to screen record myself operating viz, and I'm going to be hopelessly flailing the cursor around and not showing off the product to the best of itself. How do you do it? Let me catch you tomorrow morning before/after standup and get some tips.

tynandebold replied:
Deal :)

tynandebold reviewed (approved):
Looks great
